File: internal/scraper/types/person.go
package types

import "time"

type PersonProvider interface {
	Provider
	NewPersonProvider() PersonProvider
	SearchPerson(name string) ([]SearchData, error)
	GetPersonDetails() (PersonDetailsData, error)
}

type PersonDetailsData struct {
	Birthdate   time.Time
	Deathdate   time.Time
	Gender      int
	Description string
	Icon        string
	KnownFor    string
	ScraperInfo
}

File: legacy/app/scrapers/interfaces/person.py
from abc import ABC, abstractmethod
from dataclasses import dataclass


class PersonProvider(ABC):
    @abstractmethod
    def getPersonDetails(self):
        raise NotImplementedError()

    @abstractmethod
    def searchPerson(self, name):
        raise NotImplementedError()


@dataclass(frozen=True)
class PersonDetails:
    birthdate: str
    deathdate: str
    gender: int
    description: str
    icon: str
    knownFor: str
    scraperID: str
    scraperName: str
    scraperData: str
    scraperLink: str
